src/v/cluster/tests/data_policy_controller_test.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "cluster/data_policy_frontend.h"
+#include "cluster/errc.h"
+#include "cluster/tests/cluster_test_fixture.h"
+#include "model/metadata.h"
+#include "seastarx.h"
+#include "test_utils/fixture.h"
+#include "v8_engine/data_policy.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+#include <system_error>
+
+FIXTURE_TEST(test_change_data_policy, cluster_test_fixture) {
+    auto n1 = create_node_application(model::node_id{0});
+    auto n2 = create_node_application(model::node_id{1});
+    wait_for_controller_leadership(model::node_id{0}).get();
+    wait_for_all_members(3s).get();
+
+    model::topic_namespace topic1(test_ns, model::topic("0"));
+    model::topic_namespace topic2(test_ns, model::topic("1"));
+    model::topic_namespace topic3(test_ns, model::topic("2"));
+
+    v8_engine::data_policy dp1("1", "2");
+    auto res = n1->controller->get_data_policy_frontend()
+                 .local()
+                 .set_data_policy(topic1, dp1, 3s + model::timeout_clock::now())
+                 .get();
+    BOOST_REQUIRE_EQUAL(res, cluster::errc::success);
+
+    res = n1->controller->get_data_policy_frontend()
+            .local()
+            .set_data_policy(topic2, dp1, 3s + model::timeout_clock::now())
+            .get();
+    BOOST_REQUIRE_EQUAL(res, cluster::errc::success);
+
+    v8_engine::data_policy dp2("2", "2");
+    res = n1->controller->get_data_policy_frontend()
+            .local()
+            .set_data_policy(topic1, dp2, 3s + model::timeout_clock::now())
+            .get();
+    BOOST_REQUIRE_EQUAL(res, cluster::errc::success);
+
+    res = n1->controller->get_data_policy_frontend()
+            .local()
+            .clear_data_policy(topic2, 3s + model::timeout_clock::now())
+            .get();
+    BOOST_REQUIRE_EQUAL(res, cluster::errc::success);
+
+    res = n1->controller->get_data_policy_frontend()
+            .local()
+            .clear_data_policy(topic3, 3s + model::timeout_clock::now())
+            .get();
+    BOOST_REQUIRE_EQUAL(res, cluster::errc::data_policy_not_exists);
+
+    auto n1_map = n1->controller->get_data_policy_manager().get_current_state();
+    auto n2_map = n2->controller->get_data_policy_manager().get_current_state();
+
+    BOOST_REQUIRE_EQUAL(n1_map.size(), 1);
+    BOOST_REQUIRE_EQUAL(
+      n1_map.at(topic1).get_function_name(), dp2.get_function_name());
+    BOOST_REQUIRE_EQUAL(
+      n1_map.at(topic1).get_script_name(), dp2.get_script_name());
+    BOOST_REQUIRE_EQUAL(n2_map.size(), 1);
+    BOOST_REQUIRE_EQUAL(
+      n2_map.at(topic1).get_function_name(), dp2.get_function_name());
```

Comment:
nice! so this is testing that the data was replicated and replayed by the data policy system you added to the controller?

Reply:
Yes. I think I do not need second node, because one node can approve that cmd will be handled.